_plugins/opering_hook.rb

```diff
@@ -0,0 +1,39 @@
+require 'mkmf'
+
+OPENRING_LIST="openring/feed_list.txt"
+OPENRING_IN="openring/openring_template_in.html"
+OPENRING_OUT="_includes/openring_template_out.html"
+
+def file_was_created_more_than_x_min?(interval_min = 15)
+  file_created_at_sec = File.mtime(OPENRING_OUT).tv_sec
+  time_now_sec = Time.now.tv_sec
+
+  elapsed_time_min = (time_now_sec - file_created_at_sec) / 60
+
+  return true if elapsed_time_min >= interval_min
+  return false
+end
+
+def composing_openring_command(openring_path)
+  openring_custom_parameters = ''
+
+  File.foreach(OPENRING_LIST) do |line|
+    openring_custom_parameters += " -s #{line.chomp}"
+  end
+
+  return "#{openring_path} #{openring_custom_parameters} < #{OPENRING_IN} > #{OPENRING_OUT}"
+end
+
+Jekyll::Hooks.register :site, :post_write do
+  if file_was_created_more_than_x_min?
+    puts("-" * 80)
+
+    puts("[!] keep calm, we're running openring...")
+    openring_path = find_executable('openring')
```

Comment:
What if `openring` does not exist? You need to take action in this case, and I think this should be the first verification, even before this if.